pkg/sfu/streamtracker/interfaces.go

```diff
@@ -0,0 +1,42 @@
+package streamtracker
+
+import (
+	"fmt"
+	"time"
+)
+
+// ------------------------------------------------------------
+
+type StreamStatusChange int32
+
+func (s StreamStatusChange) String() string {
+	switch s {
+	case StreamStatusChangeNone:
+		return "none"
+	case StreamStatusChangeStopped:
+		return "stopped"
+	case StreamStatusChangeActive:
+		return "active"
+	default:
+		return fmt.Sprintf("unknown: %d", int(s))
+	}
+}
+
+const (
+	StreamStatusChangeNone StreamStatusChange = iota
+	StreamStatusChangeStopped
+	StreamStatusChangeActive
+)
+
+// ------------------------------------------------------------
+
+type StreamTrackerImpl interface {
```

Comment:
The impl that different stream tracker algorithms should implement.

Reply:
nice!